app/client/src/utils/hooks/cyanWaterbodies.ts

```diff
@@ -0,0 +1,330 @@
+  const services = useServicesContext();
+  const navigate = useNavigate();
+
+  const buildBaseLayer = useCallback(
+    (type: SublayerType) => {
+      return buildLayer(navigate, services, type, getMapView);
+    },
+    [getMapView, navigate, services],
+  );
+
+  const updateSurroundingData = useUpdateData();
+
+  // Build a group layer with toggleable boundaries
+  const { enclosedLayer, surroundingLayer } = useBoundariesToggleLayer({
+    buildBaseLayer,
+    buildFeatures,
+    enclosedFetchedDataKey: localFetchedDataKey,
+    surroundingFetchedDataKey,
+    updateLayer: updateCyanFeatureLayer,
+    updateSurroundingData,
+  });
+
+  return {
+    cyanLayer: enclosedLayer,
+    surroundingCyanLayer: surroundingLayer,
+  };
+}
+
+function useUpdateData() {
+  // Build the data update function
+  const { hucBoundaries, mapView } = useContext(LocationSearchContext);
+  const services = useServicesContext();
+  const fetchedDataDispatch = useFetchedDataDispatch();
+
+  const [hucData, setHucData] = useState<CyanWaterbodyAttributes[] | null>([]);
+  useEffect(() => {
+    const controller = new AbortController();
+
+    if (!hucBoundaries?.features?.length) {
+      setHucData([]);
+      fetchedDataDispatch({
+        type: 'success',
+        id: localFetchedDataKey,
+        payload: [],
+      });
+      return;
+    }
+
+    if (services.status !== 'success') return;
+
+    const fetchPromise = fetchCyanWaterbodies(
+      hucBoundaries.features[0].geometry,
+      services.data,
+      controller.signal,
+    );
+
+    fetchAndTransformData(
+      fetchPromise,
+      fetchedDataDispatch,
+      'cyanWaterbodies',
+    ).then((data) => setHucData(data));
+
+    return function cleanup() {
+      controller.abort();
+    };
+  }, [fetchedDataDispatch, hucBoundaries, services]);
+
+  const extent = useRef<__esri.Polygon | null>(null);
+
+  const updateSurroundingData = useCallback(
+    async (abortSignal: AbortSignal) => {
+      if (services.status !== 'success') return;
+
+      if (!mapView) return;
+
+      // Same extent, no update necessary
+      if (mapView.extent === extent.current) return;
+      extent.current = mapView.extent;
+
+      await fetchAndTransformData(
+        fetchCyanWaterbodies(
+          Polygon.fromExtent(mapView.extent),
+          services.data,
+          abortSignal,
+        ),
+        fetchedDataDispatch,
+        surroundingFetchedDataKey,
+        hucData, // Filter out HUC data
+      );
+    },
+    [fetchedDataDispatch, hucData, mapView, services],
+  );
+
+  return updateSurroundingData;
+}
+
+/*
+## Utils
+*/
+
+function buildFeatures(data: CyanWaterbodyAttributes[]) {
+  return data.map((datum) => {
+    return new Graphic({
+      attributes: datum,
+      geometry: datum.geometry,
+    });
+  });
+}
+
+function buildLayer(
+  navigate: NavigateFunction,
+  services: ServicesState,
+  type: SublayerType,
+  getMapView: () => __esri.MapView,
+) {
```

Comment:
```suggestion
function buildLayer(
  navigate: NavigateFunction,
  services: ServicesState,
  type: SublayerType,
  getMapView: () => __esri.MapView,
  getTitle: Function,
  getTemplate: Function,
) {
```

This is related to getting the Change to this location section in the surrounding features popups.